### PR TITLE
feat(router): select connector based on wallet issuer

### PR DESCRIPTION
**Pull request by vspecky**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Automatically select a connector based on the wallet issuer for Paypal (Braintree or Adyen).

### Additional Changes
None
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

We may internally process payments for different wallets through different connectors hence we should also have the logic to choose the correct connector for this purpose.

## How did you test it?

Manual, compiler-guided

## Checklist

- [x] I formatted the code `cargo +nightly fmt`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/payments/helpers.rs [hunk 1]
```diff
@@ -610,10 +610,37 @@ pub async fn get_customer_from_details(
...
+            issuer_name: api_enums::WalletIssuer::Paypal,
+            token: Some(_)
+        }))
+    ) {
+        let connector_data = api::ConnectorData::get_connector_by_name(
+            connectors,
+            "braintree",
+            api::GetToken::Connector,
+        )?;
+        Ok(api::ConnectorCallType::Single(connector_data))
```
Comment (Narayanbhat166) [inline crates/router/src/core/payments/helpers.rs:629]: Can this block just return the connector as &str and then construction of connector be done outside of this block? Code seems to be repeated here

**Review by Narayanbhat166 — CHANGES_REQUESTED** [review]
Address the slight changes requested.

**Review by SanchithHegde — APPROVED** [review]

**Review by jarnura — APPROVED** [review]
LGTM :)

**Review by SanchithHegde — APPROVED** [review]

**Comment by vspecky** [comment]
Requires further solutioning.

**Pull request closed without merging**